### PR TITLE
Skip pruning if trie log count is less than retention limit

### PR DESCRIPTION
**Pull request by siladu**
Rework error messaging in this case.

This can occur if pruning is attempted before the node has reached the configured retention limit (512 by default)

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java [hunk 1]
```diff
@@ -88,11 +95,23 @@ void prune(
...
+      } else {
+        throw new IllegalStateException(
+            "There was an error deleting the trie log backup files. Please ensure besu is working before deleting them manually.");
+      }
     } else {
-      LOG.error("Prune failed. Re-run the subcommand to load the trie logs from file.");
+      throw new IllegalStateException(
+          String.format(
+              "Remaining trie logs (%d) did not match %s (%d). Trie logs backup files have not been deleted, it is safe to rerun the subcommand.",
+              countAfterPrune, BONSAI_STORAGE_FORMAT_MAX_LAYERS_TO_LOAD, layersToRetain));
```
Comment (gfukushima) [inline besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java:114]: Should we remove BONSAI_STORAGE_FORMAT_MAX_LAYERS_TO_LOAD from this exception message?

File: besu/src/test/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelperTest.java [hunk 2]
```diff
@@ -58,7 +61,7 @@ class TrieLogHelperTest {
 
   private static final StorageProvider storageProvider = new InMemoryKeyValueStorageProvider();
   private static BonsaiWorldStateKeyValueStorage inMemoryWorldState;
-  private TrieLogHelper trieLogHelper;
+  private TrieLogHelper nonValidatingTrieLogHelper;
```
Comment (gfukushima) [inline besu/src/test/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelperTest.java:64]: what you do you mean with nonValidatingTrieLogHelper?

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java (hunk 1 above)
Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java:114]: why?

Should look like this 
> Remaining trie logs (0) did not match --bonsai-historical-block-limit (3). Trie logs backup files have not been deleted, it is safe to rerun the subcommand

File: besu/src/test/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelperTest.java (hunk 2 above)
Comment (siladu) [reply besu/src/test/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelperTest.java:64]: The validation enforces 512 minimum so I'm skipping it for the sake of the unit tests not having to create lots of data.

File: besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java (hunk 1 above)
Comment (gfukushima) [reply besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java:114]: Nevermind misread that and thought we were using the default value of that option.

**Review by gfukushima — APPROVED** [review]

**Pull request merged**